Modules/Core/Common/include/itkImageRegion.h

```diff
@@ -255,27 +251,16 @@ class ITK_TEMPLATE_EXPORT ImageRegion final : public Region
   /** Test if a continuous index is inside the region.
    * We take into account the fact that each voxel has its
    * center at the integer coordinate and extends half way
-   * to the next integer coordinate. */
+   * to the next integer coordinate, inclusive on all sides. */
   template <typename TCoordRepType>
   bool
   IsInside(const ContinuousIndex<TCoordRepType, VImageDimension> & index) const
   {
+    constexpr TCoordRepType half = 0.5;
     for (unsigned int i = 0; i < ImageDimension; ++i)
     {
-      if (Math::RoundHalfIntegerUp<IndexValueType>(index[i]) < static_cast<IndexValueType>(m_Index[i]))
-      {
-        return false;
-      }
-      // bound is the last valid pixel location
-      const auto bound = static_cast<TCoordRepType>(m_Index[i] + m_Size[i] - 0.5);
-
-      /* Note for NaN: test using negation of a positive test in order
-       * to always evaluate to true (and thus return false) when index[i]
-       * is NaN. The cast above to integer via RoundHalfIntegerUp will cast
-       * NaN into a platform-dependent value (large negative, -1 or large
-       * positive, empirically). Thus this test here is relied on
-       * to 'catch' NaN's. */
-      if (!(index[i] <= bound))
+      // Use negation of tests so that index[i]==NaN leads to returning false.
+      if (!(index[i] >= m_Index[i] - half && index[i] <= (m_Index[i] + static_cast<IndexValueType>(m_Size[i])) - half))
```

Comment:
m_Size[i] is casted here from SizeValueType to IndexValueType (so from unsigned to signed), and then converted implicitly to TCoordRepType, right? Why not directly do `static_cast<TCoordRepType>(m_Size[i])`?

Reply:
The change is a little bit worrying. There were two expressions, both returned false. Now they are combined into single expression and there is no _OR_.
**I hope that i am wrong, please ignore if you are sure.**
BTW, i have tried to replace with
`if (!(index[i] >= m_Index[i] - half) || !(index[i] <= (m_Index[i] + static_cast<IndexValueType>(m_Size[i])) - half))`
and tests passed. Are there reliable tests for the case? Please don't take my modification, it is only a very quick test. I didn't look closer.

Reply:
Seems to work well, sorry.